File: src/TradingSystem.Web/Services/PredictionApiService.cs
using System.Net.Http.Json;
using TradingSystem.Web.Models;

namespace TradingSystem.Web.Services;

public interface IPredictionApiService
{
    Task<PredictionResult?> PredictAsync(string symbol);
    Task<StrategyResult?> GetStrategyAsync(string symbol);
}

public class PredictionApiService : IPredictionApiService
{
    private readonly HttpClient _http;

    public PredictionApiService(HttpClient http)
    {
        _http = http;
    }

    public async Task<PredictionResult?> PredictAsync(string symbol)
    {
        var response = await _http.GetFromJsonAsync<ApiResponse<PredictionResult>>($"predict/{symbol}");
        return response?.Data;
    }

    public async Task<StrategyResult?> GetStrategyAsync(string symbol)
    {
        var response = await _http.GetFromJsonAsync<ApiResponse<StrategyResult>>($"strategy/{symbol}");
        return response?.Data;
    }
}

File: src/TradingSystem.Angular/src/app/services/prediction-api.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable, of } from 'rxjs';
import { map, catchError } from 'rxjs/operators';
import { ApiResponse, PredictionResult, StrategyResult } from '../models/models';

@Injectable({ providedIn: 'root' })
export class PredictionApiService {
  constructor(private http: HttpClient) {}

  predict(symbol: string): Observable<PredictionResult | null> {
    return this.http.get<ApiResponse<PredictionResult>>(`/predict/${symbol}`).pipe(
      map(r => r.data || null),
      catchError(() => of(null))
    );
  }

  getStrategy(symbol: string): Observable<StrategyResult | null> {
    return this.http.get<ApiResponse<StrategyResult>>(`/strategy/${symbol}`).pipe(
      map(r => r.data || null),
      catchError(() => of(null))
    );
  }
}
